### PR TITLE
[GRIFFIN-207] LDAP login service improvements

### PR DESCRIPTION
**Pull request by chemikadze**
- allow non-CN usernames
  - allow disabling certificate checks
  - allow limiting set of login users
  - improve logging

### CODE REVIEW

File: service/src/main/java/org/apache/griffin/core/login/ldap/SelfSignedSocketFactory.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.Socket;
+import java.net.UnknownHostException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+
+public class SelfSignedSocketFactory extends SocketFactory {
```
Comment (toyboxman) [inline service/src/main/java/org/apache/griffin/core/login/ldap/SelfSignedSocketFactory.java]: as a public class, you have to add java doc to describe

File: service/src/main/java/org/apache/griffin/core/login/ldap/SelfSignedSocketFactory.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
+package org.apache.griffin.core.login.ldap;
```
Comment (toyboxman) [inline service/src/main/java/org/apache/griffin/core/login/ldap/SelfSignedSocketFactory.java]: you should add license information here

File: service/src/main/java/org/apache/griffin/core/login/ldap/SelfSignedSocketFactory.java (hunk 1 above)
Comment (toyboxman) [inline service/src/main/java/org/apache/griffin/core/login/ldap/SelfSignedSocketFactory.java]: as a new file, you have to add java doc to describe

File: service/src/main/java/org/apache/griffin/core/login/ldap/SelfSignedSocketFactory.java (hunk 2 above)
Comment (toyboxman) [inline service/src/main/java/org/apache/griffin/core/login/ldap/SelfSignedSocketFactory.java]: you should add apache license claim here

**Comment by chemikadze** [comment]
@toyboxman i've updated the code according to your comments, thanks!

File: service/src/main/java/org/apache/griffin/core/login/ldap/SelfSignedSocketFactory.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    }
+
+    /**
+     * Part of SocketFactory contract, used by javax.net internals to create new instance.
+     */
+    public static SocketFactory getDefault() {
+        try {
+            return new SelfSignedSocketFactory();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
```
Comment (whhe) [inline service/src/main/java/org/apache/griffin/core/login/ldap/SelfSignedSocketFactory.java]: It is not recommended to throw RuntimeException directly in Griffin, maybe you can use ServiceException from GriffinException class instead.

Comment (chemikadze) [reply service/src/main/java/org/apache/griffin/core/login/ldap/SelfSignedSocketFactory.java]: @whhe fixed, thanks

**Comment by chemikadze** [comment]
@whhe @toyboxman @guoyuepeng are you good with fixes after review?

**Comment by toyboxman** [comment]
LGTM

**Pull request closed without merging**